platform/engineering/request-for-comment/tracking-submissions-from-origin-to-destination.md

```diff
@@ -0,0 +1,399 @@
+
+Now that we have criteria set, the question becomes What do we hash?
+
+#### Common Items
+
+Some items will be common between all submissions types and therefor always included in the hash.
+
+##### Submission Type
+
+Each type of submission should have a unique name, referenced in a global space. This could be a registry or a simple location where submission types could be declared/registered for common use.
+
+Defining these could be its own RFC but a basic description would be a list of Submission Types and their IDS in a tabular format:
+
+|    Submission Type     |           ID           |
+| :--------------------: | :--------------------: |
+|    VA Form 20-0995     |    va_form_20-0995     |
+|    VA Form 20-0996     |    va_form_20-0996     |
+| VA Evidence Submission | va_evidence_submission |
+|          etc           |          etc           |
+
+Ideally Submission Type IDs should be pre-normalized however this can also be handled during the normalization step of the ID generation.
+
+##### Submission Subject ID
+
+Submissions should be tied to a subject, be it a Veteran, Dependent, Representative, etc. The assumption is that all items tracked should be related to a singular subject. The submission should have some form of identifier as part of its content and we can use that identifier to both increase its uniqueness as well as deterministically identify it in a global space.
+
+Every submission is different but the assumption is that the Subject ID is considered PII by definition.
+
+Examples of Subject IDs:
+
+- Social Security Numbers (SSN)
+- Integration Control Number (ICN)
+
+This subject ID may be a combination of data in order to make it unique. This can be used in cases where there is no singular element that uniquely identifies the subject.
+
+Examples of Combination Subject IDs:
+
+- First Name + Last Name + Birth Date
+
+Subject ID does not necessarily need to be globally unique (in the above example, there can be cases where someone with the same first/last name could be born on the same date) but it has to be enough such that when combined with other data it becomes unique.
+
+#### Submission Context
+
+With the common items identified, we now turn to what makes that submission unique in the namespace of the combination of the Submission Type and Subject ID. This value will be a function of the submission type, with the subject matter experts(SME) having to define what makes that submission unique from another submission of the same type by the same subject. The SME must identify that if the same veteran submits the same form/item twice, what makes one submission unique from the other?
+
+The Submission Context can only use values present in the submission, in order to be able to regenerate that Submission ID at any point in the submission life cycle. This could mean adding data (such as a salt) to the submission if there is no way to identify a difference between two submissions. This ideally is a rare scenario.
```

Comment:
Would this need to be anything more than the unix timestamp?  How could it be guaranteed that this same salt would be inserted in all the systems?
If there is any possibility a salt would be needed, it should be the standard to avoid generation complication